make/photon/registry/builder

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+set +e
+
+echo "Usage: #./builder [registry version] [photon tag] [registry username] [registry password]"
+if [ -z $1 ]; then
+  error "Please set the 'version' variable"
+  exit 1
+fi
+if [ -z $2 ]; then
+  error "Please set the 'photonversion' variable"
+  exit 1
+fi
+if [ -z $3 ]; then
+  error "Please set the 'username' variable"
+  exit 1
+fi
+if [ -z $4 ]; then
+  error "Please set the 'password' variable"
+  exit 1
+fi
+
+VERSION="$1"
+PHOTONVERSION="$2"
+USERNAME="$3"
+PASSWORD="$4"
+PHOTONIMAGE=vmware/registry:$PHOTONVERSION
+
+set -e
+
+# the temp folder to store binary file...
+mkdir -p binary
+rm -rf binary/registry || true
+
+cd `dirname $0`
+cur=$PWD
+
+# the temp folder to store distribution source code...
+TEMP=`mktemp -d /$TMPDIR/distribution.XXXXXX`
+git clone -b $VERSION https://github.com/docker/distribution.git $TEMP
+
+echo 'build the registry binary bases on the golang:1.7.3...'
+cp Dockerfile_Golang $TEMP
+docker build -f $TEMP/Dockerfile_Golang -t registry-golang $TEMP
```

Comment:
This is a little bit, I was thinking we should provide a builder image, and build whatever source via "docker run" put the binary in volume?